cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/AbstractConfigurer.java

```diff
@@ -48,12 +51,21 @@
   protected final Id.Namespace deployNamespace;
 
   protected AbstractConfigurer(Id.Namespace deployNamespace, Id.Artifact artifactId,
-                               PluginFinder pluginFinder, PluginInstantiator pluginInstantiator) {
+                               PluginFinder pluginFinder, PluginInstantiator pluginInstantiator,
+                               @Nullable AppDeploymentRuntimeInfo runtimeInfo) {
     this.deployNamespace = deployNamespace;
     this.extraPlugins = new HashMap<>();
-    this.pluginConfigurer = new DefaultPluginConfigurer(artifactId.toEntityId(),
-                                                        deployNamespace.toEntityId(), pluginInstantiator,
-                                                        pluginFinder);
+    if (runtimeInfo != null && ClusterMode.ISOLATED.name().equals(
```

Comment:
Is there any chance the system arguments contains this cluster mode in another case? (lower, upper, etc)? If so, please make this a `equalsIgnoreCase` call.

Reply:
This argument is controlled by us and we always use ClusterMode.<>.name() to set the value, so we shouldn't worry about it. I believe this is also a common way to compare the ClusterMode in our code